SYCL/ESIMD/api/functional/ctors/ctor_broadcast.hpp

```diff
@@ -0,0 +1,196 @@
+public:
+  ConvCtorTestDescription(size_t index, DstT retrieved_val, DstT expected_val,
+                          const std::string &src_data_type,
+                          const std::string &dst_data_type)
+      : m_src_data_type(src_data_type), m_dst_data_type(dst_data_type),
+        m_retrieved_val(retrieved_val), m_expected_val(expected_val),
+        m_index(index) {}
+
+  std::string to_string() const override {
+    // TODO: Make strings for fp values more short during failure output, may be
+    // by using hex representation
+    std::string log_msg("Failed for simd<");
+
+    log_msg += m_src_data_type + ", " + std::to_string(NumElems) + ">";
+    log_msg += ", with context: " + ContextT::get_description();
+    log_msg += ", with value type: " + m_dst_data_type;
+    log_msg += ", retrieved: " + std::to_string(m_retrieved_val);
+    log_msg += ", expected: " + std::to_string(m_expected_val);
+    log_msg += ", at index: " + std::to_string(m_index);
+
+    return log_msg;
+  }
+
+private:
+  const std::string m_src_data_type;
+  const std::string m_dst_data_type;
+  const DstT m_retrieved_val;
+  const DstT m_expected_val;
+  const size_t m_index;
+};
+
+template <typename, int, typename> class Kernel;
+// The main test routine.
+// Using functor class to be able to iterate over the pre-defined data types.
+template <typename SrcT, int NumElems, typename DstT, typename TestCaseT>
+class run_test {
+
+public:
+  bool operator()(sycl::queue &queue, const std::string &src_data_type,
+                  const std::string &dst_data_type) {
+    bool passed = true;
+    const std::vector<SrcT> ref_data =
+        generate_converted_ref_data<SrcT, DstT, 1>();
+
+    for (size_t i = 0; i < ref_data.size(); ++i) {
+      passed &=
+          run_verification(queue, {ref_data[i]}, src_data_type, dst_data_type);
+    }
+
+    return passed;
+  }
+
+private:
+  bool run_verification(sycl::queue &queue, SrcT ref_val,
+                        const std::string &src_data_type,
+                        const std::string &dst_data_type) {
+    shared_vector<SrcT> result(NumElems, shared_allocator<SrcT>(queue));
+    shared_vector<SrcT> shared_ref_data(1, ref_val,
+                                        shared_allocator<SrcT>(queue));
+
+    queue.submit([&](sycl::handler &cgh) {
+      const SrcT *const ref = shared_ref_data.data();
+      SrcT *const out = result.data();
+
+      cgh.single_task<Kernel<SrcT, NumElems, DstT>>([=]() SYCL_ESIMD_KERNEL {
+        TestCaseT::template call_simd_ctor<SrcT, DstT, NumElems>(ref[0], out);
+      });
+    });
+    queue.wait_and_throw();
+
+    bool passed = true;
+    for (size_t i = 0; i < result.size(); ++i) {
+      if constexpr (std::is_same_v<SrcT, DstT>) {
+        if (!are_bitwise_equal(ref_val, result[i])) {
+          fail_test(i, result[i], ref_val, src_data_type, dst_data_type);
+        }
+      } else {
+        DstT expected = static_cast<DstT>(ref_val);
+        DstT retrieved = result[i];
```

Comment:
```suggestion
      const DstT& expected = static_cast<DstT>(ref_val);
      const DstT& retrieved = result[i];
      if constexpr (std::is_same_v<SrcT, DstT>) {
        if (!are_bitwise_equal(expected, retrieved)) {
          passed = fail_test(i, retrieved, expected, src_data_type, dst_data_type);
        }
      } else  if constexpr (...) {
```

Reply:
done in this commit: b12bfe332f0da75f23cdd49b29d10318fd4961db